shell/platform/darwin/macos/framework/Source/FLEEngine.mm

```diff
@@ -345,6 +345,38 @@ - (void)sendOnChannel:(nonnull NSString*)channel message:(nullable NSData*)messa
   }
 }
 
+- (void)sendOnChannel:(NSString*)channel
+              message:(NSData* _Nullable)message
+          binaryReply:(FlutterBinaryReply _Nullable)callback {
+  FlutterPlatformMessageResponseHandle* response_handle = nullptr;
+  struct Captures {
+    FlutterBinaryReply reply;
+  };
+  auto captures = new Captures();
+  captures->reply = callback;
+  auto message_reply = [](const uint8_t* data, size_t data_size, void* user_data) {
+    auto captures = reinterpret_cast<Captures*>(user_data);
+    NSData* reply_data = [NSData dataWithBytesNoCopy:(void*)data length:data_size freeWhenDone:NO];
```

Comment:
You need to take ownership of the data here. Currently, it is owned by the engine but it may be released after this callout. The caller could then use the data past the current scope.

Reply:
Changed to dataWithBytes:length: to copy the data.